test/hotspot/gtest/nmt/test_vmatree.cpp

```diff
@@ -415,6 +422,19 @@ TEST_VM_F(NMTVMATreeTest, SetTag) {
     tree.set_tag(75, 50, mtClass);
     expect_equivalent_form(expected, tree);
   }
+
+  {
+    testrange expected[]{
+        { 0,  50,          mtGC, si, State::Reserved},
+        {50,  75,          mtGC, si, State::Released},
+        {75, 100, mtClassShared, si, State::Reserved}
+    };
+    VMATree tree;
+    Tree::RegionData class_shared(si, mtClassShared);
+    tree.reserve_mapping(0, 100, class_shared);
+    tree.set_tag(0, 75, mtGC);
+    expect_equivalent_form(expected, tree);
+  }
```

Comment:
I think this shows an error I mentioned before, which should be fixed:

Released regions should only have mtNone as tag and no associated stack. The reason is that we don't want released sections to splinter into different parts, since that makes no sense. Released sections should always coalesce and that is easiest reached by making its properties uniformly "mtNone, no stack"

Reply:
You might be right about that. I thought that we simply ignored state and stack of released regions. Let me see what's going on.

Reply:
Ah, actually that is precisely what we see as an assert later on. `set_tag` must support address holes in some sense, otherwise we'll end up in a bad state.

Reply:
I think register_mapping should always register released sections with mtNone/no stack. Maybe even assert if you specify something different. Then, set_tag should just ignore released sections, not change anything